portality/events/consumers/application_publisher_revision_notify.py

```diff
@@ -0,0 +1,52 @@
+from flask import url_for
+
+from portality.events.consumer import EventConsumer
+from portality import constants
+from portality import models
+from portality.lib import edges
+from portality.bll import DOAJ, exceptions
+from portality.lib.seamless import SeamlessException
+
+
+class ApplicationPublisherRevisionNotify(EventConsumer):
+    ID = "application:publisher:revision:notify"
+
+    @classmethod
+    def consumes(cls, event):
+        return event.id == constants.EVENT_APPLICATION_STATUS and \
+                event.context.get("old_status") != constants.APPLICATION_STATUS_REVISIONS_REQUIRED and \
+                event.context.get("new_status") == constants.APPLICATION_STATUS_REVISIONS_REQUIRED
+
+    @classmethod
+    def consume(cls, event):
+        context = event.context
+        app_id = context.get("application")
+        if app_id is None:
+            return
+        try:
+            application = models.Application.pull(app_id)
+        except SeamlessException:
+            raise exceptions.NoSuchObjectException("Unable to locate application with id {x}".format(x=app_id))
```

Comment:
This is using the old approach, we should now be pulling the application source from the `application` context key, like

```
app_source = event.context.get("application")

        try:
            application = models.Application(**app_source)
        except Exception as e:
            raise exceptions.NoSuchObjectException("Unable to construct Application from supplied source - data structure validation error, {x}".format(x=e))
```

Reply:
Actually, there's a bit of a mix here - you have the SeamlessException but are not constructing the object from the passed data.